src/components/ResourceList/ResourceList.tsx

```diff
@@ -471,9 +475,14 @@ class ResourceList extends React.Component<CombinedProps, State> {
       <div className={styles['HeaderWrapper-overlay']} />
     ) : null;
 
-    const showEmptyState = filterControl && !this.itemsExist() && !loading;
+    const showNoResults =
+      filterControl && !this.itemsExist() && hasMoreItems && !loading;
```

Comment:
I think this is what might be the issue. `hasMoreItems` is used to create a link in the `bulkActions` when there are more results than can be displayed.  `!this.itemsExist() && hasMoreItems` should not ever be true if I'm understanding this correctly.